### PR TITLE
Feature/setup-edit-profile

### PR DESCRIPTION
**Pull request by syntialai**

### CODE REVIEW

File: features/tailormade_auth/src/main/java/com/future/tailormade_auth/feature/signUp/view/SignUpFragment.kt [hunk 1]
```diff
@@ -41,29 +42,30 @@ class SignUpFragment : BaseFragment() {
...
-    password: String, confirmPassword: String
-  ): Boolean =
-    name.isNotBlank() && email.isNotBlank() && email.isEmailValid() && birthDate.isNotBlank() && password.isNotBlank() && password.length >= 8 && confirmPassword.isNotBlank() && confirmPassword == password
+  override fun onActivityCreated(savedInstanceState: Bundle?) {
+    super.onActivityCreated(savedInstanceState)
+    showToolbar()
+  }
+
+  private fun isFormValid(name: String, email: String, birthDate: String,
+      password: String, confirmPassword: String): Boolean = name.isNotBlank() && email.isNotBlank() && email.isEmailValid() && birthDate.isNotBlank() && password.isNotBlank() && password.length >= 8 && confirmPassword.isNotBlank() && confirmPassword == password
```
Comment (blasiusneri) [inline features/tailormade_auth/src/main/java/com/future/tailormade_auth/feature/signUp/view/SignUpFragment.kt]: please add enter manually when code is expanded more than 200 char horizontally for better readability

Comment (blasiusneri) [inline features/tailormade_auth/src/main/java/com/future/tailormade_auth/feature/signUp/view/SignUpFragment.kt]: avoid magic number by put 8 in constant with (MIN_PASSWORD_LENGTH)

File: features/tailormade_profile/src/main/java/com/future/tailormade_profile/feature/editProfile/view/EditProfileFragment.kt [hunk 2]
```diff
@@ -0,0 +1,144 @@
...
+      editTextLocationEditProfile.debounceOnTextChanged(
+          viewModel.viewModelScope, viewModel::updateLocations)
+
+      textInputBirthDateEditProfile.setEndIconOnClickListener {
+        showDatePicker()
+      }
+    }
+
+    setupProfileDataObserver()
+    setupLocationObserver()
```
Comment (blasiusneri) [inline features/tailormade_profile/src/main/java/com/future/tailormade_profile/feature/editProfile/view/EditProfileFragment.kt:68]: do observe live data in onActivityCreated in fragment
https://proandroiddev.com/5-common-mistakes-when-using-architecture-components-403e9899f4cb

File: features/tailormade_profile/src/main/java/com/future/tailormade_profile/feature/editProfile/view/EditProfileFragment.kt [hunk 3]
```diff
@@ -0,0 +1,147 @@
...
+      textInputBirthDateEditProfile.error = when {
+        editTextBirthDateEditProfile.text.toString().isBlank() -> Constants.BIRTH_DATE_IS_NOT_SET
+        else -> null
+      }
+    }
+  }
+
+  private fun setupDatePicker() {
+    birthDatePicker = MaterialDatePicker.Builder.datePicker().setTitleText(
+        Constants.BIRTH_DATE_PICKER_TITLE).build()
```
Comment (blasiusneri) [inline features/tailormade_profile/src/main/java/com/future/tailormade_profile/feature/editProfile/view/EditProfileFragment.kt]: Why constant? how about put this on string?

File: features/tailormade_profile/src/main/res/layout/fragment_edit_profile_set_location.xml [hunk 4]
```diff
@@ -0,0 +1,17 @@
+<?xml version="1.0" encoding="utf-8"?>
+<layout xmlns:android="http://schemas.android.com/apk/res/android"
+    xmlns:tools="http://schemas.android.com/tools">
+
+    <LinearLayout
```
Comment (blasiusneri) [inline features/tailormade_profile/src/main/res/layout/fragment_edit_profile_set_location.xml]: use framelayout as view group for single child

File: libraries/tailormade_base/src/main/java/com/future/tailormade/util/extension/Extension.kt [hunk 5]
```diff
@@ -15,9 +24,51 @@ fun <T> Flow<T>.flowOnIO(): Flow<T> = this.flowOn(Dispatchers.IO)
...
 
+/**
+ * View extension functions
+ */
+fun EditText.debounceOnTextChanged(scope: CoroutineScope,
+    listener: (String) -> Unit) {
+  doOnTextChanged { text, _, _, count ->
+    val debounce = CoroutineHelper.debounce(scope = scope,
+        destinationFunction = listener)
+    if (count >= 3) {
```
Comment (blasiusneri) [inline libraries/tailormade_base/src/main/java/com/future/tailormade/util/extension/Extension.kt]: put 3 to constant

File: features/tailormade_profile/src/main/java/com/future/tailormade_profile/feature/editProfile/view/EditProfileFragment.kt (hunk 3 above)
Comment (syntialai) [reply features/tailormade_profile/src/main/java/com/future/tailormade_profile/feature/editProfile/view/EditProfileFragment.kt]: It used several times kak, oh use strings.xml kak?

**Review by blasiusneri — APPROVED** [review]

**Pull request merged**